src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MaxLatencySpec.groovy

```diff
@@ -72,18 +82,17 @@ class MaxLatencySpec extends HealthCheckSpecification {
         setLatencyForPaths(10, 15)
 
         when: "Create a flow with protected path, max_latency 16 and max_latency_tier_2 18"
-        def flow = flowHelperV2.randomFlow(switchPair).tap {
-            allocateProtectedPath = true
-            maxLatency = 16
-            maxLatencyTier2 = 18
-            pathComputationStrategy = PathComputationStrategy.MAX_LATENCY.toString()
-        }
-        flowHelperV2.addFlow(flow)
+        def flow = flowFactory.getBuilder(switchPair.src, switchPair.dst)
+                .withProtectedPath(true)
+                .withMaxLatency(16)
+                .withMaxLatencyTier2(18)
+                .withPathComputationStrategy(PathComputationStrategy.MAX_LATENCY)
+                .build().create()
 
         then: "Flow is created, main path is the 15 latency path, protected is 10 latency"
-        def path = northbound.getFlowPath(flow.flowId)
-        pathHelper.convert(path) == alternativePath
-        pathHelper.convert(path.protectedPath) == mainPath
+        def flowPath = flow.retrieveAllEntityPaths()
+        flowPath.getPathNodes() == alternativePath
```

Comment:
up to you: let's use params when we check both main and protected pathes together to avoid confusing
flowPath.getPathNodes(Direction.FORWARD, false) for the main path
flowPath.getPathNodes(Direction.FORWARD, true) for the protected pat

Reply:
Done